test/middleware_stack_test.rb

```diff
@@ -35,8 +35,9 @@ def test_allows_rebuilding
   end
 
   def test_allows_extending
-    build_stack Apple
+    build_handlers_stack Apple
     @conn.use Orange
+    @conn.adapter :test, &test_adapter
     assert_handlers %w[Apple Orange]
```

Comment:
I touched this test while adding the guard feature as the test started failing with the new verification. 

Originally, using the `build_stack` helper, the order of the middlewares in this test ended-up being:

```
Apple
:test (adapter)
Orange
``` 

which I assume as being invalid according to the README. I figure this is tangental to the original intent of the test itself, which was to verify the builder is mutable, so I re-wrote the test to avoid the failure.